### ISSUE TITLE
Incorrect search behaviour

### ISSUE THREAD

**Issue by CatPlanet** [hydrusnetwork/hydrus#777]
Image shows more than thousands words could.

## Environment

Hydrus version: 273
Platform: Windows
OS specifics: 7, x64

## What happens

I have multiple things going in background so it made my HDD really slow. Meanwhile Hydrus was importing images from subscription (all subs go under my personal 'import' namespace so I can tell if something was imported or added manually). I hit F5 on pane with preview of my lastest manually downloaded images and... somehow one freshly imported image got into this search. See screenshot for details.

## Steps to reproduce

1. Have HDD under heavy load
2. Have things importing under certain namespace in background
3. Have search excluding * in this particular namespace
4. Luck

## Desired outcome

Actually thing working as intended

## Error message / Log file / Screenshots

https://i.imgur.com/S6qFolV.png

**Comment by Chad90b** [comment]
> Hydrus version: 273
was this a typing mistake? if you are really on version 273, which is more than 3 years old, then unfortunately this bug report is invalid.

since version 723 (september 2017) there have been released 150 updates, many of which fix issues like this. hydrus updates more or less weekly so if you have any issues you need only upgrade.

read the help page about upgrading before you start if you are unfamiliar with the process and be careful. take backups of your .db files

**Comment by CatPlanet** [comment]
It's 423, the newest one; my bad.

**Comment by Chad90b** [comment]
ah good.

does the issue with this specific file go away if you refresh the page again now?

**Comment by CatPlanet** [comment]
Yes. After refreshing again everything was a-ok. My assumption is this bug is due to concurrency and aggressive caching of queries using namespace:* introduced in v.423

**Comment by Chad90b** [comment]
thank you for taking the time to confirm this. i believe the tag was still pending being added to the right database (you mentioned the hdd was under heavy load so this operation would be slow) when this happened. hopefully the dev can confirm this later today.

**Comment by CatPlanet** [comment]
https://i.kym-cdn.com/photos/images/original/001/453/872/091.gif
Looking forward to see what was causing it, I'm really curious.

**Comment by Chad90b** [comment]
🙏 thank you for reporting it
if you are not on the discord server ill try to update this issue with details if the dev has time to look into the tags logic today, but i am fairly confident that this is the case: just a small delay between when something is tagged and when that tag is available to the search logic

**Comment by CatPlanet** [comment]
Yes, I'm on discord and I reported this issue there first. If you need anything else just @ me here or there and I'll respond.
